### PR TITLE
Allow to get frequency spectrum from windowed trace.

### PR DESCRIPTION
**Pull request by fschlueter**
This PR is based of #735 and the diff will simplify once this is merged.

### CODE REVIEW

**Review by shallmann — COMMENTED** [review]
Implementation looks ok, requires that the user knows what he is doing to got get weird stuff...
Two potential issues:
1/ `window = [1,0,1,0,1,0,1,0,0,1,1,1]` will be totally valid, but probably not meaningful, masking a window `[1,1,1,...,1,1,0,0,0,0,0,1,1,...,1,1,1]` might also introduce weird features
2/ the other problem might need more attention: how do we get the corresponding frequencies? The get_frequencies() should also be able to take the same window then, no?

**Comment by shallmann** [comment]
@fschlueter yes that's fine with me if you add it to the get_frequencies()

**Comment by fschlueter** [comment]
@shallmann have a look :)

**Comment by shallmann** [comment]
Hmm.. why do you think the window can be `int`/`float` only? for a single value, an fft does not really make sense?

The other thing that came to my mind and might cause problems - although I did not check: The implementation works for 1D, an `electric_field` can have 3D traces, your argument is probably that for electric fields one would never set the window and hence the calculation is not required to work in > 1D?

**Comment by fschlueter** [comment]
true, what shape has a 3D efield, `nsamples, pol` or `pol, nsamles`? In the former case the code should work. The idea behind window being an int/float is that one can pass the number of samples directly. Although, I agree the name of the variable does not reflect that.

**Comment by fschlueter** [comment]
@shallmann I only test the code in my ipython shell with pseudo data but I think it should work like this. Do you like this "solution"?

File: NuRadioReco/framework/base_trace.py [hunk 1]
```diff
@@ -177,8 +195,29 @@ def add_trace_start_time(self, start_time):
...
+        """
+        if window_or_nsamples is None:
+            nsamples = self.get_number_of_samples()
+        else:
+            if isinstance(window_or_nsamples, (int, float)):
+                nsamples = int(window_or_nsamples)
+            else:
+                nsamples = int(np.sum(window_or_nsamples))
+
+        return get_frequencies(nsamples, self._sampling_rate)
```
Comment (sjoerd-bouma) [inline NuRadioReco/framework/base_trace.py:220]: I would probably have a slight preference for having the same `window` argument as in `get_frequency_spectrum`, for simplicity... but I can live with this.

File: NuRadioReco/framework/base_trace.py [hunk 2]
```diff
@@ -77,13 +77,31 @@ def get_filtered_trace(self, passband, filter_type='butter', order=10, rp=None):
...
-            self.__time_domain_up_to_date = False
-        return np.copy(self._frequency_spectrum)
+    def get_frequency_spectrum(self, window=None):
+        """
+        Returns the frequency spectrum.
+
+        Parameters
+        ----------
+        window: array of bools (default: None)
+            If not None, specifies the time window to be used for the FFT. Has to have the same length as the trace.
```
Comment (sjoerd-bouma) [inline NuRadioReco/framework/base_trace.py:87]: I'm not so sure about the term `window`, which to me implies e.g. Tukey/Hamming/etc. windows, none of which are supported here.  I suppose that's clear from the `array of bools` type specification, but maybe it's worth mentioning the word `mask` somewhere or even changing the name of the argument.

**Comment by fschlueter** [comment]
@sjoerd-bouma

**Review by sjoerd-bouma — APPROVED** [review]
I would have stuck with the simpler argument name (I'm not sure to what extent `window_mask` makes it clearer that only masks/rect windows are supported), and just added some clarification to the doc string, but I can live with this. 

Must be unrelated, but any idea why the birefringence test fails? Maybe it's not seeded?

**Pull request merged**